### PR TITLE
refactor(eventual-send): Pivot RESM to NESM

### PR DESCRIPTION
**Pull request by kriskowal**
- refactor(eventual-send): Subsume test machinery from higher layers
- refactor(eventual-send): Convert from RESM to NESM
- test(eventual-send): Add coverage machinery

### CODE REVIEW

**Review by warner — APPROVED** [review]
One down, four thousand to go

**Comment by kriskowal** [comment]
> One down, four thousand to go

The figure is more like 3 down, 37 to go 😁

File: packages/eventual-send/package.json [hunk 1]
```diff
@@ -2,12 +2,11 @@
...
   "version": "0.13.22",
   "description": "Extend a Promise class to implement the eventual-send API",
-  "parsers": {
-    "js": "mjs"
-  },
+  "type": "module",
   "main": "src/no-shim.js",
   "types": "src/index.d.ts",
   "scripts": {
+    "cover": "c8 ava",
```
Comment (michaelfig) [inline packages/eventual-send/package.json]: This script needs a specific name to be driven from the top level and ci coverage publishing.

```suggestion
    "test:c8": "c8 ava",
```

**Review by michaelfig — APPROVED** [review]
LGTM, modulo the `test:c8` change I recommend.

If there's a more conventional name than `test:c8`, please change all the other uses of it in the repo.  I'd like to see the ci coverage working in this PR.  Thanks.

File: packages/eventual-send/package.json (hunk 1 above)
Comment (kriskowal) [reply packages/eventual-send/package.json]: Integrated this, and also the options you describe in COVERAGE.md. Thanks for the detailed notes.

**Pull request merged**